File: leetcode-submissions/1652_Defuse_the_Bomb.ts
function decrypt(code: number[], k: number): number[] {
    const ans:number[]=Array.from({length:code.length},()=>0);
    if(k===0){
        return ans;
    }
    if(k>0){
        for(let i:number =0;i<code.length;i++){
            let j:number =(i+1)%code.length;
            let count :number =0;
            let temp:number =0;
            while(count <k){
                temp+=code[j];
                count ++;
                j=(j+1)%code.length;
            }
            ans[i]=temp;
        }
    }else if (k<0){
        for(let i:number =0;i<code.length;i++){
let j:number =(i-1+code.length)%code.length;
let count :number =0;
let temp:number =0;
while(count <Math.abs(k)){
    count ++;
    temp+=code[j];
    j=(j-1+code.length)%code.length;
}
ans[i]=temp;
        }
    }
    return ans;
};

File: leetcode-submissions/1652_Defuse_the_Bomb.cpp
class Solution {
public:
    vector<int> decrypt(vector<int>& code, int k) {
        vector<int>ans(code.size(),0);
        if(k==0){
            return ans ;
        }
        if(k>0){
            for(int i=0;i<code.size();i++){
                int count =0;
                int j=(i+1)%code.size();
                int temp=0;
                while(count<k){
                    count++;
                 
                   
                    temp+=code[j];
                    j=(j+1)%code.size();
                    
                }
                ans[i]=temp;
            }
        }else if(k<0){
            for(int i=0;i<code.size();i++){
                int count =0;
                int j=(i-1+code.size())%code.size();
                int temp=0;
                while(count<abs(k)){
                    count++;
                    
                    temp+=code[j];
                    j=(j-1+code.size())%code.size();
                }
                ans[i]=temp;
            }
        }
        return ans;
    }
};